### PR TITLE
feat: Add auth via DEX OpenID

### PR DESCRIPTION
**Pull request by tumido**
Resolves: https://github.com/operate-first/apps/issues/65
Requies: https://github.com/operate-first/apps/pull/248

Implements base secrets for Argo, Grafana, Grafana2 and Hue for now (for a quicker onboarding, the apps need to turn on SSO anyways)

### CODE REVIEW

File: auth/base/dex-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          - http://URL/oauth2/callback
+        secretEnv: ARGO_SECRET
+
+      - id: grafana
+        name: Grafana
+        redirectURIs:
+          - https://URL/login/generic_oauth
+        secretEnv: GRAFANA_SECRET
+
+      - id: grafana_observatorium
```
Comment (4n4nd) [inline auth/base/dex-cm.yaml:38]: @tumido can we remove observatorium grafana from this? It already has login disabled and we don't need it to be enabled.

Comment (tumido) [reply auth/base/dex-cm.yaml:38]: I'm cool with that, np, I'll remove it

**Review by HumairAK — APPROVED** [review]
/lgtm
/approve

Let's try it!

**Pull request merged**